felix/fv/bpf_dual_stack_test.go

```diff
@@ -208,116 +209,138 @@ func describeBPFDualStackTests(ctlbEnabled bool) bool {
-			})
-
-			It("Should connect to w[0][0] via clusterIP (IPv4 and IPv6)", func() {
-				port := uint16(testSvc.Spec.Ports[0].Port)
-				cc.ExpectSome(w[1][0], TargetIP(clusterIPs[0]), port)
-				cc.ExpectSome(w[1][0], TargetIP(clusterIPs[1]), port)
-
-				cc.ExpectSome(w[0][1], TargetIP(clusterIPs[0]), port)
-				cc.ExpectSome(w[0][1], TargetIP(clusterIPs[1]), port)
-				cc.CheckConnectivity()
-			})
-
-			It("Should connect to w[0][0] via nodePort (IPv4 and IPv6)", func() {
-				cc.ExpectSome(w[1][0], TargetIP(felixIP(0)), npPort)
-				cc.ExpectSome(w[0][1], TargetIP(felixIP(0)), npPort)
+		if ipv6Dataplane {
+			Context("with a policy allowing ingress to w[0][0] from all regular workloads", func() {
+				var (
+					testSvc          *v1.Service
+					testSvcNamespace string
+				)
+
+				npPort := uint16(30333)
+				clusterIPs := []string{"10.101.0.10", "dead:beef::abcd:0:0:10"}
+				JustBeforeEach(func() {
+					tc.TriggerDelayedStart()
+					ensureAllNodesBPFProgramsAttached(tc.Felixes)
+				})
+				BeforeEach(func() {
+					k8sClient = infra.(*infrastructure.K8sDatastoreInfra).K8sClient
+					_ = k8sClient
+					testSvc = k8sServiceForDualStack("test-svc", clusterIPs, w[0][0], 80, 8055, int32(npPort), "tcp")
+					testSvcNamespace = testSvc.ObjectMeta.Namespace
+					_, err := k8sClient.CoreV1().Services(testSvcNamespace).Create(context.Background(), testSvc, metav1.CreateOptions{})
+					Expect(err).NotTo(HaveOccurred())
+					Eventually(k8sGetEpsForServiceFunc(k8sClient, testSvc), "10s").Should(HaveLen(1),
+						"Service endpoints didn't get created? Is controller-manager happy?")
+				})
+				It("Should connect to w[0][0] from all other workloads with IPv4 and IPv6", func() {
+					cc.ExpectSome(w[0][1], w[0][0])
+					cc.ExpectSome(w[1][0], w[0][0])
+					cc.ExpectSome(w[1][1], w[0][0])
+
+					cc.Expect(Some, w[0][1], w[0][0], ExpectWithIPVersion(6))
+					cc.Expect(Some, w[1][0], w[0][0], ExpectWithIPVersion(6))
+					cc.Expect(Some, w[1][1], w[0][0], ExpectWithIPVersion(6))
+					cc.CheckConnectivity()
+				})
+
+				It("Should connect to w[0][0] via clusterIP (IPv4 and IPv6)", func() {
+					port := uint16(testSvc.Spec.Ports[0].Port)
+					cc.ExpectSome(w[1][0], TargetIP(clusterIPs[0]), port)
+					cc.ExpectSome(w[1][0], TargetIP(clusterIPs[1]), port)
+
+					cc.ExpectSome(w[0][1], TargetIP(clusterIPs[0]), port)
+					cc.ExpectSome(w[0][1], TargetIP(clusterIPs[1]), port)
+					cc.CheckConnectivity()
+				})
+
+				It("Should connect to w[0][0] via nodePort (IPv4 and IPv6)", func() {
+					cc.ExpectSome(w[1][0], TargetIP(felixIP(0)), npPort)
+					cc.ExpectSome(w[0][1], TargetIP(felixIP(0)), npPort)
+
+					cc.ExpectSome(w[1][0], TargetIP(felixIP6(0)), npPort)
+					cc.ExpectSome(w[0][1], TargetIP(felixIP6(0)), npPort)
+					cc.CheckConnectivity()
+				})
 
-				cc.ExpectSome(w[1][0], TargetIP(felixIP6(0)), npPort)
-				cc.ExpectSome(w[0][1], TargetIP(felixIP6(0)), npPort)
-				cc.CheckConnectivity()
 			})
+			if !ctlbEnabled {
```

Comment:
is this specific to not having ctlb enabled or you just want to run these tests once and that is enough? Perhaps worth a comment in the code?